bot/seasons/evergreen/space.py

```diff
@@ -0,0 +1,236 @@
+        rover: Optional[str] = "curiosity"
+    ) -> None:
+        """
+        Get random Mars image by date. Support both SOL (martian solar day) and earth date and rovers.
+
+        Earth date formatting is YYYY-MM-DD. Use `.space mars dates` to get all currently available rovers.
+        """
+        rover = rover.lower()
+        if rover not in self.rovers:
+            await ctx.send(
+                (
+                    f"Invalid rover `{rover}`.\n"
+                    f"**Rovers:** `{'`, `'.join(f'{r.capitalize()}' for r in self.rovers)}`"
+                )
+            )
+            return
+
+        # When date not provided, get random SOL date between 0 and rover's max.
+        if date is None:
+            date = random.randint(0, self.rovers[rover]["max_sol"])
+
+        params = {}
+        if isinstance(date, int):
+            params["sol"] = date
+        else:
+            params["earth_date"] = date.date().isoformat()
+
+        result = await self.fetch_from_nasa(f"mars-photos/api/v1/rovers/{rover}/photos", params)
+        if len(result["photos"]) < 1:
+            err_msg = (
+                f"We can't find result in date "
+                f"{date.date().isoformat() if isinstance(date, datetime) else f'{date} SOL'}.\n"
+                f"**Note:** Dates must match with rover's working dates. Please use `{ctx.prefix}space mars dates` to "
+                "see working dates for each rover."
+            )
+            await ctx.send(err_msg)
+            return
+
+        item = random.choice(result["photos"])
+        await ctx.send(
+            embed=await self.create_nasa_embed(
+                f"{item['rover']['name']}'s {item['camera']['full_name']} Mars Image", "", item["img_src"],
+            )
+        )
+
+    @mars.command(name="dates", aliases=["d", "date", "rover", "rovers", "r"])
+    async def dates(self, ctx: Context) -> None:
+        """Get current available rovers photo date ranges."""
+        await ctx.send("\n".join(
+            f"**{r.capitalize()}:** {i['min_date']} **-** {i['max_date']}" for r, i in self.rovers.items()
+        ))
+
+    async def fetch_from_nasa(
+        self,
+        endpoint: str,
+        additional_params: Optional[Dict[str, Any]] = None,
+        base: Optional[str] = NASA_BASE_URL,
+        use_api_key: bool = True
+    ) -> Dict[str, Any]:
+        """Fetch information from NASA API, return result."""
+        params = {}
+        if use_api_key:
+            params["api_key"] = Tokens.nasa
+
+        # Add additional parameters to request parameters only when they provided by user
+        if additional_params is not None:
+            params.update(additional_params)
+
+        async with self.http_session.get(url=f"{base}/{endpoint}?{urlencode(params)}") as resp:
+            return await resp.json()
+
+    async def create_nasa_embed(self, title: str, description: str, image: str, footer: Optional[str] = "") -> Embed:
```

Comment:
This function does not need to be `async` at all, consider turning it back to be a normal sync method.